src/shared/reboot-util.c

```diff
@@ -107,3 +125,63 @@ int shall_restore_state(void) {
 
         return r > 0 ? ret : true;
 }
+
+static int xen_kexec_loaded(void) {
+#if HAVE_XENCTRL
+        size_t size;
+        _cleanup_close_ int privcmd_fd = -1;
+        _cleanup_close_ int buf_fd = -1;
+        void *buffer;
+        xen_kexec_status_t *status;
+        int r;
+
+        size = page_size();;
+        if (sizeof(*status) > size)
```

Comment:
status is not initialized here, maybe use the type instead? It looks a bit weird

Reply:
`sizeof` only evaluates the type. This form is usually preferred as it remains valid even if the type of a var changes.